code/modules/research/xenobiology/xenobiology.dm

```diff
@@ -188,6 +188,18 @@
 	origin_tech = "biotech=6"
 	var/list/not_interested = list()
 	var/sentience_type = SENTIENCE_ORGANIC
+	var/heat_stage = 0 //When used at stage 2 or above, the Sentience Potion explodes. Stop bugging ghosts
+
+/obj/item/slimepotion/sentience/examine(mob/user)
+	. = ..()
+	if(user.Adjacent(src))
+		switch(heat_stage)
+			if(0)
+				return
+			if(1)
+				. += "<span class='warning'>The vial is hot to the touch.</span>"
+			if(2)
+				. += "<span class='warning'>The vial is scalding hot! Is it really a good idea to use this..?</span>"
```

Comment:
```suggestion
	if(!user.Adjacent(src))
		return
	switch(heat_stage)
		if(0)
			return
		if(1)
			. += "<span class='warning'>The vial is hot to the touch.</span>"
		if(2)
			. += "<span class='warning'>The vial is scalding hot! Is it really a good idea to use this..?</span>"
```
This could be an early return,